world/physics/models/solar_generation.py

```diff
@@ -0,0 +1,232 @@
+                                                                  [0, -1, 0],
+                                                                  [0, 0, 1]])))  # -y face
+        cube_surfaces.append(cube_surfaces[2].transform(np.array([[1, 0, 0],
+                                                                  [0, 1, 0],
+                                                                  [0, 0, -1]])))  # -z face
+
+        """
+        deployables_tilt_angle
+                |  /    deployables_dir
+                | /            ^
+        ________|/             |
+        |       |              |
+        |       |   --> deployables_offset_dir
+        |_______|
+        deployables_tilt_dir is into the page
+        """
+        deployables_tilt_dir = np.roll(deployables_dir, 1)  # doesn't matter which of the 4 perpendiculars we choose
+        deployables_offset_dir = np.cross(deployables_tilt_dir, deployables_dir)
+        deployables_tilt = Rotation.from_rotvec(deployables_tilt_angle * deployables_tilt_dir).as_matrix()
+
+        deployable_surfaces = [Surface(is_solar_panel=True,
+                                       center=R * (np.eye(3) + deployables_tilt) @ deployables_dir + R * deployables_offset_dir,
+                                       normal=deployables_tilt @ deployables_offset_dir,
+                                       x_dir=deployables_tilt_dir,
+                                       y_dir=deployables_tilt @ deployables_dir,
+                                       width=0.1,
+                                       height=0.1)]
+
+        # add the 3 other rotated copies of the deployable surfaces
+        rot_90 = Rotation.from_rotvec((np.pi / 2) * deployables_dir).as_matrix()
+        for i in range(0, 4):
+            deployable_surfaces.append(deployable_surfaces[0].transform(rot_90 ** i))
+
+        # add solar panels on the other side of the extendable panels
+        deployable_surfaces.extend([deployable_surface.flip_normal() for deployable_surface in deployable_surfaces])
+        return cube_surfaces + deployable_surfaces
+
+    @staticmethod
+    def get_intersections(surface: Surface, ray_starts: np.ndarray, ray_dir: np.ndarray) -> np.ndarray:
+        """
+        Check if a set of rays intersect a Surface.
+
+        :param surface: A Surface object representing the surface.
+        :param ray_starts: A numpy array of shape (n, 3) representing the starting points of the rays in the body frame.
+        :param ray_dir: A 3-element array representing the direction of the rays in the body frame.
+        :return: A boolean array of shape (n,) indicating whether the corresponding rays intersect the surface.
+        """
+        cos_theta = np.dot(surface.normal, ray_dir)
+        if np.abs(cos_theta) < 1e-3:
+            # The rays are parallel to the surface.
+            return np.zeros(ray_starts.shape[0], dtype=bool)
+
+        ts = (surface.center - ray_starts) @ surface.normal / cos_theta
```

Comment:
Need a help file for this portion. Took me a long time to get what this was doing. Image with the surface, a vector starting from ray_start parallel to ray_dir and finding out where the expected intersection is would help a lot.